doc

justinmk opened the pull request:
- Lua
- developer guidelines
- `MAINTAIN.md`
- TUI: cleanup
- TUI: mention Windows terminfo builtins
- cleanup `if_pyth`, redirect `python-bindeval` tag

bfredl commented on src/nvim/memline_defs.h:
```diff
@@ -26,9 +26,17 @@ typedef struct ml_chunksize {
+/// Nodes are stored on ml_mfp (memfile_T):
+///   pointer_block: internal nodes
+///   data_block: leaf nodes
+/// Memline also has "chunks" of 800 lines that are separate from the 128-tree structure ?
```
Primarily used to speed up `line2byte()` and `byte2line()` if I understood it correctly.

justinmk committed:
doc [ci skip] #9478

- Lua
- developer guidelines
- MAINTAIN.md
- TUI: cleanup
- TUI: mention Windows terminfo builtins
- cleanup if_pyth, redirect python-bindeval tag

Helped-by: Björn Linse <bjorn.linse@gmail.com>
Helped-by: erw7 <erw7.github@gmail.com>

justinmk commented on runtime/doc/term.txt:
```diff
@@ -42,34 +43,34 @@ unlike most other environment variables.
-  Linux virtual terminal      linux, linux-256color               Y
+  tmux                        tmux, tmux-256color                 Y
+  vtpcon                      vtpcon                              Y
+  Windows console (cmd.exe)   win32con                            Y
```
@erw7 I could not find any info on "vtpcon". Does it have a full name? Is there a URL that describes it?

erw7 replied:
`vtpcon` is the name I named irresponsibly, because vim calls "on MS-Windows console: support for 'termguicolors'" as `vtp`. Also, `win32con` named it because &term is `win32` on vim. "cmd.exe" also works on `ConEmu`, so I think "cmd.exe" is not appropriate. I think that it is better to say "conhost.exe" on the terminal, but what about it?

+ `vtpcon` is a "conhost.exe" that supports `ENABLE_VIRTUAL_TERMINAL_PROCESSING`
+ `win32con` is "conhost.exe" which does not support `ENABLE_VIRTUAL_TERMINAL_PROCESSING`.

erw7 replied:
Since terminal emulation are provided by `libuv`  rather than `conhost`, `libuv` or `uvtty` may be more appropriate than `win32con`.

justinmk replied:
> Also, win32con named it because `&term` is win32 on vim. 

@erw7 	should we use the same TERM name (`win32`) in that case?

erw7 replied:
In the case of vim, tgc can be used when `&term=win32` and `has('vcon')`, so neovim does not behave like vim, but it may be better to change the TERM name from `win32con` to `win32` and from `vtpcon` to `vcon`.

justinmk commented on runtime/doc/term.txt:
```diff
@@ -42,34 +43,34 @@ unlike most other environment variables.
+  anything libvte-based       vte, vte-256color                   Y
+   (e.g. GNOME Terminal)      (aliases: gnome, gnome-256color)
+  ConEmu                      conemu                              Y
+  cygwin                      cygwin                              Y
```
@erw7 Is this for cygwin pty object, or when is it used? I assume it is not used for mintty, which I think typically sets `TERM=xterm-256color` .

erw7 replied:
Use cygwin when terminal is `conhost`(It is when you use `Cygwin.bat` from `cmd.exe`.).

erw7 replied:
Supplement. Using `TERM=cygwin` is for nvim compiled with Cygwin.

justinmk committed:
gen_api_vimdoc.py: Do not wrap on hyphens, long words

- Any long symbol is intentional and should never be hardwrapped.
- Vim help tags are often hyphenated, and hardwrapping on hyphens breaks
  the Vim help syntax parser.

justinmk commented on runtime/doc/api.txt:
```diff
@@ -400,26 +442,26 @@ nvim_set_current_dir({dir})                           *nvim_set_current_dir()*
 
 nvim_get_current_line()                              *nvim_get_current_line()*
-                Gets the current line
+                Gets the current line.
```
@neovim/core 	Even though it seems to be a doxygen(?) convention to omit the period in the first sentence, it does not make sense for us, because in contexts such as Vim help docs it looks like a mistake. 

Also it's inconsistent: we terminate almost every other docstring, even sentence fragments in parameter descriptions ...
